site/app/models/gradeable/AutoGradedVersion.php

```diff
@@ -0,0 +1,439 @@
+<?php
+
+namespace app\models\gradeable;
+
+
+use app\exceptions\FileNotFoundException;
+use app\libraries\Core;
+use app\libraries\DateUtils;
+use app\libraries\FileUtils;
+use app\libraries\GradingQueue;
+use app\models\AbstractModel;
+
+/**
+ * Class AutoGradedVersion
+ * @package app\models\gradeable
+ *
+ * Data about and results of autograding for one submission version
+ *
+ * @method int getVersion()
+ * @method float getNonHiddenNonExtraCredit()
+ * @method float getNonHiddenExtraCredit()
+ * @method float getHiddenNonExtraCredit()
+ * @method float getHiddenExtraCredit()
+ * @method \DateTime getSubmissionTime()
+ * @method bool isAutogradingComplete()
+ */
+class AutoGradedVersion extends AbstractModel {
+    /** @var GradedGradeable Reference to the GradedGradeable */
+    private $graded_gradeable = null;
+    /** @property @var int The submission version for this AutoGradedVersion */
+    protected $version = 0;
+    /** @property @var float The number of "normal" points */
+    protected $non_hidden_non_extra_credit = 0;
+    /** @property @var float The number of "normal extra credit" points */
+    protected $non_hidden_extra_credit = 0;
+    /** @property @var float The number of "hidden" points */
+    protected $hidden_non_extra_credit = 0;
+    /** @property @var float The number of "hidden extra credit" points */
+    protected $hidden_extra_credit = 0;
+    /** @property @var \Datetime Time the user submitted this version */
+    protected $submission_time = null;
+    /** @property @var bool If the autograding has complete for this version */
+    protected $autograding_complete = false;
```

Comment:
If you're going to do `AutoGradedVersion` you should probably do `$auto_grading_complete`

Reply:
Its kind of a mess, but `Autograding` is long-form.  `AutoGraded` is shorthand for `AutogradingGraded`

Reply:
Except "Autograded" is also past tense of "to Autograde" and is spelled with both capses all over the place... AuGhh